src/include/network/routines.rb

```diff
@@ -711,36 +712,44 @@ def GetAllInterfaces
       end
     end
 
-    def SetLinkUp(dev_name)
-      log.info("Setting link up for interface #{dev_name}")
-      Run("ip link set #{dev_name} up")
+    # Calls wicked ifup if it is the currenct backend, set the link up
+    # otherwise
+    #
+    # @param [String] name of interface to put down
+    def SetIfaceUp(dev_name)
+      log.info("Setting interface #{devname} up")
+      Run("ifup #{dev_name} up")
     end
 
-    def SetLinkDown(dev_name)
-      log.info("Setting link down for interface #{dev_name}")
-      Run("ip link set #{dev_name} down")
+    # Calls wicked ifdown if it is the current backend, set the link down
+    # otherwise
+    #
+    # @param [String] name of interface to put down
+    def SetIfaceDown(dev_name)
+      log.info("Setting interface #{devname} down")
+      Run("ifdown #{dev_name}")
     end
 
-    # Tries to set all available interfaces up
+    # Tries to set all interfaces up
     #
     # @return [boolean] false if some of interfaces cannot be set up
-    def SetAllLinksUp
+    def SetAllIfacesUp
       interfaces = GetAllInterfaces()
 
       return false if interfaces.empty?
 
-      interfaces.all? { |i| SetLinkUp(i) }
+      interfaces.all? { |i| SetIfaceUp(i) }
```

Comment:
When wicked is the backend maybe could be interested to call `wicked ifup interfaces.join(" ")` from a performance point of view.